packages/extension/src/ui/features/accountTokens/tokens.state.ts

```diff
@@ -337,17 +338,28 @@ async function fetchFungibleTokenFromFullNode(network: Network, tokenId: string)
       return undefined
     }
 
-    const metadata = await fetchImmutable(`${tokenId}-token-metadata`, () => nodeProvider.fetchFungibleTokenMetaData(tokenId))
-    const token: Token = {
-      id: tokenId,
-      networkId: network.id,
-      name: Buffer.from(metadata.name, 'hex').toString('utf8'),
-      symbol: Buffer.from(metadata.symbol, 'hex').toString('utf8'),
-      decimals: metadata.decimals,
-      verified: false
+    const metadata = await fetchImmutable(`${tokenId}-token-metadata`, async () => {
+      try {
+        return (await nodeProvider.fetchFungibleTokenMetaData(tokenId))
+      } catch (e: any) {
+        if (e.message.startsWith('Failed to call contract')) {
+          return { name: undefined, symbol: undefined, decimals: 0 }
+        } else {
+          throw e
+        }
```

Comment:
It should be okey since this is for fungible tokens.

Reply:
Will the wallet keep fetching? For example, if there is an assertion error in the code

Reply:
For other exceptions it will not be cached, so next time when we display fungible tokens, it will try to make requests again.

We want to avoid the situation where the error is transient and we inadvertently cached the token as invalid, and we know `Failed to call contract` error is persistent. Please let me know if there are better ways to do this.

Reply:
I think the general practice is to fetch the data using backoff strategies. If the data can't be fetched multiple times, then it's considered unavailable, unless the error is 429. SWR handles 429 differently from other errors in our configuration.

Reply:
Yes, exceptions will be subject to the SWR retry strategy. For fetching fungible token metadata specifically, [current SWR setup](https://github.com/alephium/extension-wallet/blob/master/packages/extension/src/ui/features/accountTokens/tokens.state.ts#L207-L212) is to retry if the error is 429 using SWR's default retry strategy, and not retry for other types of exceptions.

In this case, we are handling a known and persistent exception for getting token metadata and cache it as invalid, such that when user call it again it wont make more requests, I guess this is on top of the SWR retry strategy. WDTY?